x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/ExtractedFieldsDetector.java

```diff
@@ -253,14 +253,13 @@ private void removeFieldsWithIncompatibleTypes(Set<String> fields, Set<FieldSele
         Iterator<String> fieldsIterator = fields.iterator();
         while (fieldsIterator.hasNext()) {
             String field = fieldsIterator.next();
-            if (hasCompatibleType(field) == false) {
-                addExcludedField(field, "unsupported type; supported types are " + getSupportedTypes(), fieldSelection);
-                fieldsIterator.remove();
-            }
```

Comment:
sorry I didn't catch this in review. two if statements without a break...should have been obvious 🤦

Reply:
Yeah, I did my fair share of facepalming too.